test/org/zaproxy/zap/extension/pscanrulesAlpha/UserControlledJavascriptEventScannerUnitTest.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class UserControlledJavascriptEventScannerUnitTest extends PassiveScannerTest {
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new UserControlledJavascriptEventScanner();
+	}
+
+	public HttpMessage createMessage() {
+		HttpMessage msg = new HttpMessage();
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		try {
+			requestHeader.setURI(new URI("http://example.com/i.php", false));
+		} catch (URIException | NullPointerException e) {
+		}
+		requestHeader.setMethod(HttpRequestHeader.GET);
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+		msg.getResponseHeader().setStatusCode(HttpStatusCode.OK);
+		msg.getResponseHeader().addHeader(HttpHeader.CONTENT_TYPE, "text/html");
+		return msg;
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseIsNotStatusOk() {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getResponseHeader().setStatusCode(HttpStatusCode.NOT_ACCEPTABLE);
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseIsNotHtml() {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getResponseHeader().setHeader(HttpHeader.CONTENT_TYPE, "application/json");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestHasNoGetParams() {
+		// Given
+		HttpMessage msg = createMessage();
+		// WHen
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestParamsHaveNoValues() throws URIException, NullPointerException {

```

Comment:
Could throw `Exception` instead (same for other tests here and in other classes).

Reply:
Is that preferable?